src/WebJobs.Script/Rpc/LanguageWorkerChannelInitializationService.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    public class LanguageWorkerChannelInitializationService : IHostedService
+    {
+        private readonly IEnumerable<string> _languages = new List<string>()
+        {
+            LanguageWorkerConstants.JavaLanguageWorkerName
+        };
+
+        private readonly IEnumerable<string> _languagesSupportedOnLinux = new List<string>();
+
+        private readonly string _currentLanguageWorkerRuntime;
+
+        private ILanguageWorkerChannelManager _languageWorkerChannelManager;
+
+        public LanguageWorkerChannelInitializationService(ILanguageWorkerChannelManager languageWorkerChannelManager)
```

Comment:
Doesn't DI prevent this service from being null? Do you need the explicit check/throw?

Reply:
Yes. DI should take care. When testing early prototype, this was null if DI container did find all the services LangaugeWorkerChannelManager depended on. Leaving it here as it is safer.